crates/subspace-farmer/src/dsn.rs

```diff
@@ -1,15 +1,99 @@
-use std::ops::Range;
+use std::ops::{ControlFlow, Range};
+use std::sync::{Arc, Mutex};
 
-use futures::Stream;
-use subspace_core_primitives::PieceIndexHash;
+use futures::{Stream, StreamExt};
+
+use subspace_core_primitives::{FlatPieces, PieceIndex, PieceIndexHash, PublicKey, Sha256Hash};
+use subspace_solving::PieceDistance;
 
 use crate::PiecesToPlot;
 
+// TODO: Should it be a distinct type which implements arithmetics operations?
+pub type PieceIndexHashNumber = PieceDistance;
```

Comment:
`PieceDistance` is definitely the wrong type here. Should be just 256-bit number. Maybe call it that - `U256`.

Reply:
I can use `uint` crate again and generate another type for that

Reply:
`PieceDistance` should probably be renamed into `U256` and then `type PieceDistance = U256;`.